### PR TITLE
[PM-4375] Reuse show file selector callout logic

### PR DESCRIPTION
**Pull request by aj-rosado**
## Type of change

Separating the logic from the `FilePopoutCalloutComponent` into a new Service

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

The logic used on the `FilePopoutCalloutComponent` is also necessary to Parent Components. We are separating that logic into a service so that it can be used by all the components.

## Code changes

- **services.module.ts:** Added the new `FilePopoutUtilsService`
- **file-popout-callout.component.ts:** Removed the logic regarding which callout should be shown and using the new service that deals with that.
- **send-add-edit.component.ts:** Using the `FilePopoutUtilsService` to check if the callout is visible
- **file-popout-utils.service.ts:** Service containing all the logic if the callout should be visible and which text should present to the user

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

File: apps/browser/src/tools/popup/services/file-popout-utils.service.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import { Injectable } from "@angular/core";
+
+import { PlatformUtilsService } from "@bitwarden/common/platform/abstractions/platform-utils.service";
+
+import { PopupUtilsService } from "../../../popup/services/popup-utils.service";
+
+@Injectable()
+export class FilePopoutUtilsService {
```
Comment (audrey-jensen) [inline apps/browser/src/tools/popup/services/file-popout-utils.service.ts:8]: ⚠️ Public and protected exported members should have documentation explaining their purpose and arguments.

**Review by audrey-jensen — CHANGES_REQUESTED** [review]
Overall this looks great! It just needs documentation.

File: apps/browser/src/tools/popup/services/file-popout-utils.service.ts (hunk 1 above)
Comment (aj-rosado) [reply apps/browser/src/tools/popup/services/file-popout-utils.service.ts:8]: Added documentation

Comment (audrey-jensen) [reply apps/browser/src/tools/popup/services/file-popout-utils.service.ts:8]: The constructor and the class itself also require documentation.

👍🏻 What you've already written is fantastic!

Comment (aj-rosado) [reply apps/browser/src/tools/popup/services/file-popout-utils.service.ts:11]: Added the missing documentation

File: apps/browser/src/tools/popup/services/file-popout-utils.service.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+  private isLinux(win: Window): boolean {
+    return win?.navigator?.userAgent.indexOf("Linux") !== -1;
+  }
+
+  private isUnsupportedMac(win: Window): boolean {
+    return (
+      this.platformUtilsService.isChrome() && win?.navigator?.appVersion.includes("Mac OS X 11")
+    );
+  }
```
Comment (audrey-jensen) [inline apps/browser/src/tools/popup/services/file-popout-utils.service.ts:75]: 💭 Looking more closely at this, I wonder whether the user agent/navigator queries should be moved into the platform service.

```ts
private isUnsupportedMac(win: Window): boolean {
    return (
      this.platformUtilsService.isChrome() && 
      this.platformUtilsService.isMac() && 
      !this.platormUtilsService.isSupportedOS()
    );
  }
```

We can keep this conversation going outside of the PR. It's not a blocking issue.

**Review by audrey-jensen — APPROVED** [review]

**Review by djsmith85 — APPROVED** [review]
The changes are looking good.

**Pull request merged**